plugins/09_Shadowlands/zones/the_maw.lua

```diff
@@ -1581,6 +1581,315 @@ map.nodes[85375524] = Class('Tormentor', ns.node.Node, {
+            52033, -- No Soul Left Behind
+            52034, -- Snail Stomping
+            52035, -- Just Don't Ask Me to Spell It
+            52036, -- Double Dromans
+            52037, -- That's Going to Sting
+            52038, -- The Soul Blade
+            52039, -- A Shady Place
+            52040 -- Heavy Handed Tactics
+        }
+    }), -- Rooting Out the Evil
+    ns.reward.Spacer(), Achievement({
+        id = 15044,
+        criteria = {
+            52078, -- Elder Gwenna
+            52079, -- Foreman Thorodir
+            52080, -- Te'zan
+            52081, -- Warden Casad
+            52082, -- Kivarr
+            52083 -- Guardian Kota
+        }
+    }), -- Krrprripripkraak's Heroes
+    ns.reward.Spacer(), Section('{item:185991}'),
+    Mount({item = 186000, id = 1476}), -- Wild Hunt Legsplitter
+    Pet({item = 186547, id = 3116}) -- Invasive Buzzer
+}
+
+local KYRIAN_ASSAULT_REWARDS = {
+    Achievement({id = 15000, criteria = 51723}), -- United Front
+    ns.reward.Spacer(), Achievement({
+        id = 15004,
+        criteria = ({
+            id = 1,
+            qty = true,
+            suffix = L['kyrian_assault_quantity_note1']
+        })
+    }), -- A Sly Fox
+    ns.reward.Spacer(), Achievement({
+        id = 15034,
+        criteria = {
+            52010, -- Mine's Bigger
+            52011, -- Heart and Soul,
+            52012, -- No One Floats Down Here
+            52013, -- Encouraging Words
+            52014, -- Courage of the Soul
+            52015, -- Saved By The Bells
+            52016, -- United In Pride
+            52041, -- The Ember Count
+            52042, -- Kill The Flame
+            52043 -- The Dreadful Blend
+        }
+    }), -- Wings Against the Flames
+    ns.reward.Spacer(), Achievement({
+        id = 15041,
+        criteria = ({
+            id = 1,
+            qty = true,
+            suffix = L['kyrian_assault_quantity_note2']
+        })
+    }), -- The Zovaal Shuffle
+    ns.reward.Spacer(), Section('{item:185993}'),
+    Pet({item = 186546, id = 3103}), -- Copperback Etherwyrm
+    Toy({item = 187185}) -- Vesper of Faith
+}
+
+local ASSAULTS = {
+    [63543] = NECROLORD_ASSAULT_REWARDS,
+    [63822] = VENTHYR_ASSAULT_REWARDS,
+    [63823] = NIGHT_FAE_ASSAULT_REWARDS,
+    [63824] = KYRIAN_ASSAULT_REWARDS
+}
+
+local assaultHandled = false
+
+hooksecurefunc(GameTooltip, 'Show', function(self)
+    if assaultHandled then return end
+    local owner = self:GetOwner()
+    if owner and owner.questID then
+        local rewards = ASSAULTS[owner.questID]
+        if rewards and #rewards > 0 then
+            for i, reward in ipairs(rewards) do
+                if reward:IsEnabled() then reward:Render(self) end
+            end
+            assaultHandled = true
+            self:AddLine(' ') -- add blank line after rewards
+            self:Show()
+        end
+    end
+end)
+
+hooksecurefunc(GameTooltip, 'ClearLines',
+    function(self) assaultHandled = false end)
```

Comment:
This code block is supposed to add reward information to the current assault on The Maw map.

There is chance that if you looked at it during the Venthyr assault it may have had the incorrect quest ID. Wowhead has two quests for that assault. If that wasn't the problem then I am not sure since it works on my end.

I noticed today that your `ns.covenants` includes assault quests so I implemented a reference to that table instead of just manually typing the values.

Anyway, here is a screenshot of what it should do.

Instead of having two pins that contain the same information (our Handynotes pin and the current assault quest if incomplete) it queries the map to see if the quest is active and then hides our Handynotes pin if it is.

Unfortunately, I also learned today that if you are in a different zone but navigate to The Maw map then it will show the Handynotes pin AND the tooltip augmentation on the quest pin. Must be a limitation of `C_QuestLog.IsOnMap()` that it only looks at the map for where the player is. I couldn't figure out any other way to detect if an assault quest was active. Maybe you have a better idea.

![Screenshot-(14)](https://user-images.githubusercontent.com/88683851/175799251-3bc52ea8-3fb1-4951-9522-7db049767ea1.png)